vault/core.go

```diff
@@ -708,6 +708,12 @@ func (c *Core) sealInitCommon(req *logical.Request) (retErr error) {
 		return retErr
 	}
 
+	// Create an identifier for the request
+	var err error
+	req.ID, err = uuid.GenerateUUID()
```

Comment:
I think we should do this as early as possible -- in `http/logical.go` in the `buildLogicalRequest` function. That way we can refer to it at any point in time, even if that's still in the http code. Right now we're only exposing it for auditing, but that could change in the future.